### PR TITLE
Fix Shed Tail rounding down

### PR DESCRIPTION
**Pull request by adrivrie**
The amount of damage the user of Shed Tail takes is 50% of its max hp rounded up.

https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-9436537

### CODE REVIEW

File: data/moves.ts [hunk 1]
```diff
@@ -16458,13 +16458,13 @@ export const Moves: {[moveid: string]: MoveData} = {
 				this.add('-fail', source, 'move: Shed Tail');
 				return this.NOT_FAIL;
 			}
-			if (source.hp <= source.maxhp / 2 || source.maxhp === 1) { // Shedinja clause
+			if (source.hp <= (source.maxhp + 1) / 2) { // rounds up
```
Comment (Karthik99999) [inline data/moves.ts:16461]:
```suggestion
			if (source.hp <= Math.ceil(source.maxhp / 2) || source.maxhp === 1) { // Shedinja clause
```

File: data/moves.ts [hunk 2]
```diff
@@ -16458,13 +16458,13 @@ export const Moves: {[moveid: string]: MoveData} = {
...
 				this.add('-fail', source, 'move: Shed Tail', '[weak]');
 				return this.NOT_FAIL;
 			}
 		},
 		onHit(target) {
-			this.directDamage(target.maxhp / 2);
+			this.directDamage((target.maxhp + 1) / 2); // rounds up
```
Comment (Karthik99999) [inline data/moves.ts:16467]:
```suggestion
			this.directDamage(Math.ceil(target.maxhp / 2));
```

File: data/moves.ts (hunk 1 above)
Comment (adrivrie) [reply data/moves.ts]: Turned the +1 stuff (I was thinking of integer division) to Math.ceil, but I think the shedinja clause should not return.

**Comment by Karthik99999** [comment]
The description should be updated to say "rounded half up"

**Comment by DaWoblefet** [comment]
Thanks adrivrie!

**Pull request merged**